lib/model/settings.dart

```diff
@@ -376,8 +421,13 @@ class GlobalSettingsStore extends ChangeNotifier {
   /// A value of null means the setting will revert to following
   /// the app's default.
   ///
+  /// The setting's value will be changed only if new value is different
+  /// from the old one.
+  ///
   /// See also [getBool].
   Future<void> setBool(BoolGlobalSetting setting, bool? value) async {
+    if (value == getBool(setting)) return;
```

Comment:
This doesn't seem like it has the right behavior if the setting isn't currently set.
* If `value` is null, this will go ahead and redundantly unset the (already unset) setting.
* But if `value` equals whatever value is currently the default, this will abort even though the logic below would make a real change in the database.

Instead let's use `_boolData`, so as to operate at the same layer as the rest of this method below.

Reply:
This should also get some tests, as it's a change to some of our core data-managing logic.